src/Aquarius.Client/Samples/Client/WebServiceExceptionHandler.cs

```diff
@@ -9,6 +9,9 @@ public static class WebServiceExceptionHandler
     {
         public static SamplesApiException CreateSamplesApiExceptionFromResponse(WebServiceException e)
         {
+            if (e.IsAny500())
+                return new SamplesMaintenanceModeException($"{e.Message}: AQUARIUS Samples is in maintenance mode", e);
```

Comment:
You cannot assume that Any 500 from the server means maintenance mode. This would incorrectly report maintenance for any number of bugs.

The best compromise I can think of is to make a call to the status api in this case, and if it fails report maintenance. You'll need a bit more logic in this function to prevent infinite recursion (don't call status if the error is already for the status api).